pkg/sql/schemachanger/scplan/internal/rules/helpers.go

```diff
@@ -372,6 +369,11 @@ func isIndexDependent(e scpb.Element) bool {
 	return false
 }
 
+func isCheckConstraint(e scpb.Element) bool {
```

Comment:
Wouldn't this be a good place for a TODO to expand this predicate to include all non-index-backed constraints?